src/client/dfuse/ops/read.c

```diff
@@ -58,44 +57,39 @@ dfuse_cb_read(fuse_req_t req, fuse_ino_t ino, size_t len, off_t position, struct
 		mock_read = true;
 	}
 
-	D_ALLOC(buff, len);
-	if (!buff)
-		D_GOTO(err, rc = ENOMEM);
+	if (len > ev->de_iov.iov_buf_len) {
+		D_ERROR("Fuse read buffer not large enough %zx > %zx\n", len,
```

Comment:
why is this an error if no error is returned?

Reply:
With fuse you can specify max_read_size or pass 0 at init time to mean "as large as the kernel supports", as dfuse passes 0 it has no way of knowing what the maximum request size will be so it hard-codes to 1mb.  If this error happens then there's better kernel support and we need to change the buffer size but the code will still work as expected.  The error is only here to highlight to us if the kernel limit ever does change.

Reply:
then INFO or maybe WARN would be more appropriate in this case?

Reply:
Updated to WARN and I've added a comment to the source.

I didn't want to use INFO as we want to know and change dfuse if the kernel value ever changes.